lib/rust/ensogl/core/src/display/render/passes/cache_shapes.rs

```diff
@@ -83,14 +90,32 @@ impl pass::Definition for CacheShapesPass {
         self.layer.camera().update(&self.scene);
         self.scene.display_object.update(&self.scene);
         self.layer.update();
+        // The asynchronous update of the scene's display object initiated above will eventually
+        // set our layer's camera's transformation. Handle the camera update when all
+        // previously-initiated FRP events finish being processed.
+        let handle = frp::microtasks::next_microtask({
+            let camera = self.layer.camera();
+            let scene = self.scene.clone_ref();
+            let camera_ready = Rc::clone(&self.camera_ready);
+            move || {
+                camera.update(&scene);
+                camera_ready.set(true);
+            }
+        });
```

Comment:
Are we not updating cameras in Scene::update as well? If I remember correctly we do. How this code relates to the update functions in Scene? Everything is probably fine, but I do not understand yet why we are handling the same thing in two places (or maybe my knowledge about the (EnsoGL) world is not up to date now ;)

Reply:
The update finishes asynchronously now, due to the `batch` FRP used in layer updates. Other layers work OK because we poll their dirty flags unconditionally when processing all updates, so they eventually see the results of all microtasks--but this layer is only updated once, and its draws only occur once, so we need to know when it's ready.